code/game/objects/items/weapons/implants/implant_explosive.dm

```diff
@@ -26,16 +27,20 @@
 		activate("death")
 
 /obj/item/implant/explosive/activate(cause)
-	if(!cause || !imp_in)	return 0
+	if(!cause || !imp_in)
+		return FALSE
 	if(cause == "action_button" && alert(imp_in, "Are you sure you want to activate your microbomb implant? This will cause you to explode!", "Microbomb Implant Confirmation", "Yes", "No") != "Yes")
-		return 0
+		return FALSE
+	if(detonating)
+		return FALSE
```

Comment:
As far as I can tell, `trigger()` is only called here when the player makes an emote:
https://github.com/ParadiseSS13/Paradise/blob/dc9ce6dad727e9ad26a85a2b812ce09a6bec63e3/code/modules/mob/living/carbon/human/emote.dm#L21-L23
Activating it through the action button is done separately here:
https://github.com/ParadiseSS13/Paradise/blob/dc9ce6dad727e9ad26a85a2b812ce09a6bec63e3/code/game/objects/items/weapons/implants/implant.dm#L23-L24
Both of those call `activate()`, so it makes more sense to just have the check in there.

Reply:
Yeah alright, in that case it would be better to stick at the top of the proc:
```dm
/obj/item/implant/explosive/activate(cause)
	if(!cause || !imp_in || detonating)
		return FALSE
```

Reply:
Needs to stay after the `alert()`, since if you spam click the action button it'll run the proc a bunch of times up until the alert and stop there until the user presses a button.
Previously if you pressed "Yes" on the other prompts after you'd already exploded it would cause a runtime since the holder had been gibbed, but I added the `detonating` check to avoid that.

Reply:
You're right I forgot you can just ignore pop ups and still click/interact with stuff. *Flashback to the multiple guardians bug*

Reply:
*Yep*